### PR TITLE
Fix Strategy Video Convert

### PR DESCRIPTION
**Pull request by oscarlopez8**

### CODE REVIEW

File: src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java [hunk 1]
```diff
@@ -81,13 +81,14 @@
...
             @RequestParam (defaultValue = "0") int height, @RequestParam (defaultValue = "") String videoCodec,
             @RequestParam (defaultValue = "") String audioCodec, @RequestParam (defaultValue = "") String videoBitRate,
             @RequestParam (defaultValue = "") String audioBitRate, @RequestParam (defaultValue = "-1") int quality,
             @RequestParam (defaultValue = "0") int channelsNumber, @RequestParam (defaultValue = "") String volume,
-            @RequestParam (defaultValue = "") short rotate) throws CommandValueException {
+            @RequestParam (defaultValue = "") short rotate, @RequestParam (defaultValue = "") short thumbNail)
+            throws CommandValueException {
 
         Md5Checksum md5Checksum = new Md5Checksum();
         Param param = new VideoParam("thirdparty\\FFmpeg\\bin\\ffmpeg.exe");
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java]: here you can use PathJcf class for getting ffmpeg binary.

File: src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java [hunk 2]
```diff
@@ -125,6 +126,7 @@ public String videoConverter(
                 videoParam.setAudioCodec(audioCodec);
                 videoParam.setVideoBitRate(videoBitRate);
                 videoParam.setAudioBitRate(audioBitRate);
+                videoParam.setThumbnail(thumbNail);
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java]: it is correct **thumbNail** or thumbnail?

File: src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoRotate.java [hunk 3]
```diff
@@ -22,14 +22,11 @@
     // Content value to rotate.
     private short commandValue;
 
-    // Content value 90.
-    private final short degrees90 = 90;
+    // Content value Comma.
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoRotate.java]: change this comment to **Content Comma value**

File: src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoRotate.java [hunk 4]
```diff
@@ -46,10 +43,20 @@ public CommandVideoRotate(short commandValue) {
...
+            case 180:
+                return SPACE + VideoCommand.VF.getCommand() + SPACE + this.quotationMark +
+                        VideoCommand.ROTATE.getCommand() + this.COMMA + VideoCommand.ROTATE.getCommand() +
+                        this.quotationMark;
+            case 270:
+                return SPACE + VideoCommand.VF.getCommand() + SPACE + this.quotationMark +
+                        VideoCommand.ROTATE.getCommand() + this.COMMA + VideoCommand.ROTATE.getCommand() + this.COMMA +
+                        VideoCommand.ROTATE.getCommand() + this.quotationMark;
+            default:
+                return null;
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoRotate.java]: Use exceptions, in this command class

File: src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoThumbNail.java [hunk 5]
```diff
@@ -45,8 +45,8 @@ public CommandVideoThumbNail(int commandValue) {
     @Override
     public String command() {
         if (commandValue > numberZero) {
-            return SPACE + VideoCommand.THUMBNAIL.getCommand() + SPACE + commandValue +
-                  VideoCommand.V_FRAMES + second;
+            return this.SPACE + VideoCommand.THUMBNAIL.getCommand() + this.SPACE + secondBegin + this.SPACE +
+                  VideoCommand.V_FRAMES.getCommand() + this.SPACE + commandValue;
         }
         return null;
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoThumbNail.java]: use exceptions in this command class.

File: src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java [hunk 6]
```diff
@@ -59,17 +59,18 @@ public FileResult convert(Param param) {
         try {
             List<ICommandStrategy> list = new ArrayList<>();
             list.add(new CommandFFMpegPath());
-            list.add(new CommandFFMpeg());
+            //list.add(new CommandFFMpeg());
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java]: if you are not using this line of code, you can deleted please.

File: src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java [hunk 7]
```diff
@@ -59,17 +59,18 @@ public FileResult convert(Param param) {
...
-            list.add(new CommandVideoAspectRatio(Integer.toString(videoParam.getAspectRatio())));
-            list.add(new CommandVideoScale(videoParam.getWidth(), videoParam.getHeight()));
-            list.add(new CommandVideoConverter());
-            list.add(new CommandVideoThumbNail(Integer.parseInt(videoParam.getThumbnail())));
-            list.add(new CommandVideoRotate(videoParam.getRotate()));
+            //list.add(new CommandVideoAspectRatio(videoParam.getAspectRatio()));
+            //list.add(new CommandVideoScale(videoParam.getWidth(), videoParam.getHeight()));
+            //list.add(new CommandVideoConverter());
+            //list.add(new CommandVideoThumbNail(videoParam.getThumbnail()));
+            //list.add(new CommandVideoRotate(videoParam.getRotate()));
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java]: if you are not using this lines of code, you can deleted.

File: src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java [hunk 8]
```diff
@@ -59,17 +59,18 @@ public FileResult convert(Param param) {
...
             list.add(new CommandVideoFrameRate(videoParam.getFrameRate()));
             list.add(new CommandOutputFilePath(videoParam.getOutputPathFile()));
             list.add(new CommandOutputFileName(videoParam.getOutputFileName()));
             String stringCommand = getCommand(list);
+            System.out.println(stringCommand);
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java]: delete this line of code is not necessary.

File: src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java [hunk 9]
```diff
@@ -78,6 +79,9 @@ public FileResult convert(Param param) {
 
             }
             process.waitFor();
+            if (videoParam.getThumbnail() > 0) {
+
+            }
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java]: this conditional is valid?

File: src/main/java/com/jalasoft/jfc/model/strategy/CommandFFMpegPath.java [hunk 10]
```diff
@@ -20,18 +25,35 @@
 public class CommandFFMpegPath implements ICommandStrategy {
 
     // Content command value.
-    private final String FFMPEG_PATH = "thirdparty/ffmpeg/bin/ffmpeg.exe";
+    private  String FFMpegPath;
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/strategy/CommandFFMpegPath.java]: documment this variable.

File: src/main/java/com/jalasoft/jfc/model/strategy/CommandOutputFileName.java [hunk 11]
```diff
@@ -39,7 +39,7 @@ public CommandOutputFileName(String commandValue) {
      * @throws CommandValueException
      */
     public String command() throws CommandValueException {
-        String regexRule = "[^a-zA-Z0-9]";
+        String regexRule = "[^a-zA-Z0-9.]";
         String replaceValue = "";
 
             if (commandValue.equals("")) {
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/strategy/CommandOutputFileName.java]: check identation in this line. And you can use commandValue.isempty()

File: src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java [hunk 12]
```diff
@@ -46,55 +43,61 @@
...
+        int returnValue = -1;
+        try {
+            Process process = Runtime.getRuntime().exec(stringCommand);
+            process.waitFor();
+            returnValue = process.exitValue();
+        } catch (InterruptedException | IOException e) {
+            e.printStackTrace();
+        }
+        return returnValue;
+    }
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java]: improve this method
-become void this method

File: src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java [hunk 13]
```diff
@@ -46,55 +43,61 @@
     private final static Logger LOGGER = Logger.getLogger(VideoConverter.class.getName());
 
     /**
-     * This method convert a video format to another format.
+     * Runs string command.
+     * @param stringCommand value of command.
+     * @return 0 when the process was executed successfully.
+     */
+    private int runCommand(String stringCommand){
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java]: add CommandValueException

File: src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java [hunk 14]
```diff
@@ -46,55 +43,61 @@
     private final static Logger LOGGER = Logger.getLogger(VideoConverter.class.getName());
 
     /**
-     * This method convert a video format to another format.
+     * Runs string command.
+     * @param stringCommand value of command.
+     * @return 0 when the process was executed successfully.
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java:48]: add throws documentation.

File: src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java [hunk 15]
```diff
@@ -46,55 +43,61 @@
...
+            e.printStackTrace();
+        }
+        return returnValue;
+    }
+
+    /**
+     * This method converts a Video to other format.
      * @param param
      * @return FileResult object or null value.
-     * @throws IOException
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java:53]: change to CommandValueException

File: src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java [hunk 16]
```diff
@@ -46,55 +43,61 @@
...
      */
-    public FileResult convert(Param param) {
-
-        VideoParam videoParam = (VideoParam) param;
+    public FileResult convert(Param param) throws CommandValueException, IOException {
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java]: delete IOException

File: src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java [hunk 17]
```diff
@@ -46,55 +43,61 @@
...
+    }
 
+    /**
+     * This method is for getting the string command.
+     *
+     * @param param
+     * @return command concatenated.
+     * @throws CommandValueException
+     */
+    public String getCommand(Param param) throws CommandValueException, IOException {
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java]: delete IOException

File: src/main/java/com/jalasoft/jfc/model/video/VideoParam.java [hunk 18]
```diff
@@ -60,7 +60,7 @@
     private short rotate;
 
     // Content a specific time.
-    private String thumbnail;
+    private short thumbNail;
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/video/VideoParam.java]: this is correct thumbNail? or thumbnail

Comment (oscarlopez8) [reply src/main/java/com/jalasoft/jfc/model/video/VideoParam.java]: thumbNail

File: src/main/java/com/jalasoft/jfc/model/IConverter.java [hunk 19]
```diff
@@ -33,5 +33,5 @@
      * @throws ConvertException this.
      */
 
-    FileResult convert(Param param) throws ConvertException, CommandValueException;
+    FileResult convert(Param param) throws ConvertException, CommandValueException, IOException;
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/IConverter.java]: delete IOException

File: src/main/java/com/jalasoft/jfc/model/video/VideoParam.java [hunk 20]
```diff
@@ -60,7 +60,7 @@
     private short rotate;
 
     // Content a specific time.
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/video/VideoParam.java:62]: **Content a specific time value.**

File: src/main/java/com/jalasoft/jfc/model/video/VideoParam.java [hunk 21]
```diff
@@ -291,10 +291,10 @@ public void setRotate(short rotate) {
 
     /**
      * Sets thumbnail value.
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/video/VideoParam.java]: **Sets thumbnail's value.**

File: src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java [hunk 22]
```diff
@@ -104,6 +104,7 @@ public String videoConverter(
             Files.write(path, bytes);
             videoParam.setInputPathFile(path.toString());
             md5FileUploaded = md5Checksum.getMd5(path.toString());
+            System.out.println(md5FileUploaded);
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java]: delete this line

File: src/main/java/com/jalasoft/jfc/model/strategy/CommandFFMpegPath.java [hunk 23]
```diff
@@ -27,16 +27,12 @@
     // Content command value.
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/strategy/CommandFFMpegPath.java]: improve this comment to Content ffmpeg command value.

File: src/main/java/com/jalasoft/jfc/model/strategy/CommandFFMpegPath.java [hunk 24]
```diff
@@ -27,16 +27,12 @@
...
     private  String FFMpegPath;
 
-    // Variable type PathJfc.
-    PathJfc pathJfc;
-
     /**
      * This initialize PathJfc and gets the FfMpeg Path.
      * @throws IOException when is a invalid file.
      */
     public CommandFFMpegPath() throws IOException {
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/strategy/CommandFFMpegPath.java]: change this type of Exception. Also in its documentation.

File: src/main/java/com/jalasoft/jfc/model/strategy/CommandFFMpegPath.java [hunk 25]
```diff
@@ -27,16 +27,12 @@
...
-        pathJfc = new PathJfc();
-        FFMpegPath = pathJfc.getMagickPath();
+        FFMpegPath = PathJfc.getFfmpegPath();
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/strategy/CommandFFMpegPath.java]: document this variable

File: src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoBitRate.java [hunk 26]
```diff
@@ -0,0 +1,44 @@
+package com.jalasoft.jfc.model.strategy;
+
+import com.jalasoft.jfc.model.exception.CommandValueException;
+import com.jalasoft.jfc.model.video.VideoCommand;
+
+/**
+ * Class changes Aspect Ratio.
+ *
+ * @version 0.1 02 Ene 2020.
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoBitRate.java]: change month to english

File: src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoBitRate.java [hunk 27]
```diff
@@ -0,0 +1,44 @@
...
+ * @author Oscar Lopez.
+ */
+public class CommandVideoBitRate implements ICommandStrategy {
+
+    // Content command value.
+    private String commandValue;
+
+    /**
+     * Creates a new CommandVideoBitRate object.
+     *
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoBitRate.java]: delete this separations in methods

File: src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoBitRate.java [hunk 28]
```diff
@@ -0,0 +1,44 @@
...
+     * @param commandValue, receive a value.
+     */
+    public CommandVideoBitRate(String commandValue) {
+        this.commandValue = commandValue;
+    }
+
+    /**
+     * This builds a command.
+     *
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoBitRate.java]: delete this separation lines in methods documentation

File: src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoCodec.java [hunk 29]
```diff
@@ -0,0 +1,43 @@
...
+
+    /**
+     * This builds a command.
+     *
+     * @return command concatenated.
+     */
+    @Override
+    public String command() throws CommandValueException {
+        try {
+            if (commandValue != "") {
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoCodec.java]: here you can use if (!commandValue.isEmpty())

File: src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoFormat.java [hunk 30]
```diff
@@ -0,0 +1,38 @@
...
+     */
+    public CommandVideoFormat(String commandValue) {
+        this.commandValue = commandValue;
+    }
+
+    /**
+     * Builds a command.
+     * @return String of a command.
+     */
+    public String command() throws CommandValueException {
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoFormat.java]: document the exception

File: src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoFrameRate.java [hunk 31]
```diff
@@ -37,7 +38,14 @@ public CommandVideoFrameRate(String commandValue) {
      * @return command concatenated.
      */
     @Override
-    public String command() {
-        return this.SPACE + VideoCommand.FRAME_RATE.getCommand() + SPACE + commandValue;
+    public String command() throws CommandValueException {
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoFrameRate.java]: document exception

File: src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoFrameRate.java [hunk 32]
```diff
@@ -37,7 +38,14 @@ public CommandVideoFrameRate(String commandValue) {
...
+        try {
+            if (commandValue != "") {
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoFrameRate.java]: here you can use if (!commandValue.isEmpty())

File: src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoScale.java [hunk 33]
```diff
@@ -46,11 +47,16 @@ public CommandVideoScale(int width, int height) {
      * @return command concatenated.
      */
     @Override
-    public String command() {
-        if (width > numberZero && height > numberZero) {
-            return this.SPACE + VideoCommand.VF.getCommand() + this.SPACE + VideoCommand.SCALE.getCommand() +
-                   width + VideoCommand.COLON.getCommand() + height;
+    public String command() throws CommandValueException {
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoScale.java]: document exception

File: src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoThumbNail.java [hunk 34]
```diff
@@ -20,20 +20,20 @@
...
 
     // Content value to rotate.
-    private int commandValue;
+    private boolean commandValue;
 
     // Content value of one second to extract a thumbnail.
-    private final String secondBegin = "00:00:01.000";
+    private final String secondBegin = "00:00:02.000";
 
     //Content number 0
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoThumbNail.java]: update this comment

File: src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoThumbNail.java [hunk 35]
```diff
@@ -20,20 +20,20 @@
...
-    private final short numberZero = 0;
+    private final short getFrames = 100;
 
     /**
      * Creates a new CommandVideoRotate object.
      *
      * @param commandValue, receive a value.
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/strategy/CommandVideoThumbNail.java]: you van improve this comment to @param commandValue, **receive boolean a value**

File: src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java [hunk 36]
```diff
@@ -60,37 +63,47 @@ private int runCommand(String stringCommand){
     }
 
     /**
-     * This method converts a Video to other format.
+     * It is converts a Video to other format.
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java]: It converts a Video to another format.

File: src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java [hunk 37]
```diff
@@ -60,37 +63,47 @@ private int runCommand(String stringCommand){
...
      * @param param
      * @return FileResult object or null value.
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java:62]: update this comment

File: src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java [hunk 38]
```diff
@@ -60,37 +63,47 @@ private int runCommand(String stringCommand){
...
      */
     public FileResult convert(Param param) throws CommandValueException {
         FileResult fileResult = new FileResult();
-        String stringCommand = getCommand(param);
+        VideoParam videoParam = (VideoParam)param;
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java:67]: document the exception

File: src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java [hunk 39]
```diff
@@ -60,37 +63,47 @@ private int runCommand(String stringCommand){
...
+        StringBuilder stringCommand = new StringBuilder();
+
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java:68]: delete this line

File: src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java [hunk 40]
```diff
@@ -60,37 +63,47 @@ private int runCommand(String stringCommand){
...
         return fileResult;
     }
 
     /**
-     * This method is for getting the string command.
+     * It is for getting the string command.
      *
      * @param param
      * @return command concatenated.
      * @throws CommandValueException
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java]: could you improve this comment?

File: src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java [hunk 41]
```diff
@@ -60,37 +63,47 @@ private int runCommand(String stringCommand){
...
-            list.add(new CommandVideoScale(videoParam.getWidth(), videoParam.getHeight()));
-            list.add(new CommandVideoConverter());
-            list.add(new CommandVideoThumbNail(videoParam.getThumbnail()));
-            list.add(new CommandVideoRotate(videoParam.getRotate()));
-            list.add(new CommandVideoFrameRate(videoParam.getFrameRate()));
+            //list.add(new CommandVideoAspectRatio(videoParam.getAspectRatio()));
+            //list.add(new CommandVideoScale(videoParam.getWidth(), videoParam.getHeight()));
+            //list.add(new CommandVideoConverter());
+            //list.add(new CommandVideoRotate(videoParam.getRotate()));
+            //list.add(new CommandVideoCodec(videoParam.getVideoCodec()));
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java]: if you are not using this lines of code, delete it please

File: src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java [hunk 42]
```diff
@@ -102,4 +115,23 @@ public String getCommand(Param param) throws CommandValueException {
             throw new CommandValueException(e.getMessage(), this.getClass().getName());
         }
     }
+
+    public String getThumbnail(Param param) throws CommandValueException {
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java:126]: this methos needs documentation

File: src/main/java/com/jalasoft/jfc/model/video/VideoFormat.java [hunk 43]
```diff
@@ -0,0 +1,40 @@
+package com.jalasoft.jfc.model.video;
+
+/**
+ * This enum is used for specify types of image formats.
+ *
+ * @version 0.1 16 Dic 2019
+ *
+ * @author Alan Escalera
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/video/VideoFormat.java]: update this information please

File: src/main/java/com/jalasoft/jfc/model/strategy/CommandFFMpegPath.java (hunk 25 above)
Comment (oscarlopez8) [reply src/main/java/com/jalasoft/jfc/model/strategy/CommandFFMpegPath.java]: It's documented on line 27

File: src/main/java/com/jalasoft/jfc/model/strategy/CommandFFMpegPath.java [hunk 44]
```diff
@@ -24,20 +24,20 @@
...
         FFMpegPath = PathJfc.getFfmpegPath();
     }
 
     /**
      * Generates a command.
-     * @return exe of ImageMagick path.
+     * @return exe of FFMpeg path.
      */
     @Override
     public String command() throws CommandValueException {
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/strategy/CommandFFMpegPath.java]: document CommandValueException

File: src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandFFMpegPath.java [hunk 45]
```diff
@@ -11,7 +11,12 @@
 
 import com.jalasoft.jfc.model.command.ICommandStrategy;
```
Comment (Amet015) [inline src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandFFMpegPath.java]: delete this line please

File: src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoAspectRatio.java [hunk 46]
```diff
@@ -9,9 +9,13 @@
 
 package com.jalasoft.jfc.model.command.ffmpeg;
 
+
```
Comment (Amet015) [inline src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoAspectRatio.java]: Delete this lines

File: src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoBitRate.java [hunk 47]
```diff
@@ -0,0 +1,45 @@
+package com.jalasoft.jfc.model.command.ffmpeg;
+
```
Comment (Amet015) [inline src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoBitRate.java:2]: delete this line please

File: src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoCodec.java [hunk 48]
```diff
@@ -0,0 +1,44 @@
+package com.jalasoft.jfc.model.command.ffmpeg;
+
```
Comment (Amet015) [inline src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoCodec.java:2]: delete this line please

File: src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoBitRate.java [hunk 49]
```diff
@@ -0,0 +1,45 @@
...
+ * @author Oscar Lopez.
+ */
+public class CommandVideoBitRate implements ICommandStrategy {
+
+    // Content command value.
+    private String commandValue;
+
+    /**
+     * Creates a new CommandVideoBitRate object.
+     *
```
Comment (Amet015) [inline src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoBitRate.java]: Delete this line

File: src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoBitRate.java [hunk 50]
```diff
@@ -0,0 +1,45 @@
...
+     * @param commandValue, receive a value.
+     */
+    public CommandVideoBitRate(String commandValue) {
+        this.commandValue = commandValue;
+    }
+
+    /**
+     * This builds a command.
+     *
```
Comment (Amet015) [inline src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoBitRate.java]: delete this line

File: src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoCodec.java [hunk 51]
```diff
@@ -0,0 +1,44 @@
...
+ * @author Oscar Lopez.
+ */
+public class CommandVideoCodec implements ICommandStrategy {
+
+    // Content command value.
+    private String commandValue;
+
+    /**
+     * Creates a new CommandVideoCodec object.
+     *
```
Comment (Amet015) [inline src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoCodec.java:21]: delete this line

File: src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoCodec.java [hunk 52]
```diff
@@ -0,0 +1,44 @@
...
+     * @param commandValue, receive a value.
+     */
+    public CommandVideoCodec(String commandValue) {
+        this.commandValue = commandValue;
+    }
+
+    /**
+     * This builds a command.
+     *
```
Comment (Amet015) [inline src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoCodec.java:30]: delete this line

File: src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoConverter.java [hunk 53]
```diff
@@ -11,6 +11,9 @@
 
 import com.jalasoft.jfc.model.command.ICommandStrategy;
```
Comment (Amet015) [inline src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoConverter.java:13]: delete this line

File: src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoFrameRate.java [hunk 54]
```diff
@@ -11,6 +11,9 @@
 
 import com.jalasoft.jfc.model.command.ICommandStrategy;
```
Comment (Amet015) [inline src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoFrameRate.java:13]: delete this line

File: src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoRotate.java [hunk 55]
```diff
@@ -9,6 +9,8 @@
 
 package com.jalasoft.jfc.model.command.ffmpeg;
 
+
```
Comment (Amet015) [inline src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoRotate.java:12]: delete this line

File: src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoScale.java [hunk 56]
```diff
@@ -9,6 +9,8 @@
 
 package com.jalasoft.jfc.model.command.ffmpeg;
 
+
```
Comment (Amet015) [inline src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoScale.java:12]: delete this line

File: src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoThumbNail.java [hunk 57]
```diff
@@ -22,20 +22,25 @@
...
+    private boolean commandValue;
 
-    // Content value of one second to extract a thumbnail.
-    private final short second = 1;
+    // Content value of two second to extract a thumbnail.
+    private final String secondBegin = "00:00:02.000";
 
-    //Content number 0
-    private final short numberZero = 0;
+    //Content number 4
```
Comment (Amet015) [inline src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoThumbNail.java]: add piriod .

File: src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoThumbNail.java [hunk 58]
```diff
@@ -22,20 +22,25 @@
...
+    private final short second = 4;
 
+    //Content number 10
```
Comment (Amet015) [inline src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoThumbNail.java]: add piriod

File: src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoThumbNail.java [hunk 59]
```diff
@@ -22,20 +22,25 @@
...
+    private final short framePerSecond = 10;
+
+    //Content a thumbnail size
```
Comment (Amet015) [inline src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoThumbNail.java]: add period .

File: src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoThumbNail.java [hunk 60]
```diff
@@ -22,20 +22,25 @@
...
+    private final String thumbnailSize = "128:128";
     /**
      * Creates a new CommandVideoRotate object.
      *
```
Comment (Amet015) [inline src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoThumbNail.java]: delete this line

File: src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java [hunk 61]
```diff
@@ -28,12 +29,9 @@
 import com.jalasoft.jfc.model.command.ContextStrategy;
 import com.jalasoft.jfc.model.command.ICommandStrategy;
 
-import java.io.BufferedReader;
 import java.io.IOException;
```
Comment (Amet015) [inline src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java:32]: add new line

File: src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java [hunk 62]
```diff
@@ -44,61 +42,101 @@
  */
 public class VideoConverter implements IConverter {
 
-    private final static Logger LOGGER = Logger.getLogger(VideoConverter.class.getName());
+    /**
+     * Runs string command.
+     * @param stringCommand value of command.
+     * @return 0 when the process was executed successfully.
+     */
+    private void runCommand(String stringCommand){
```
Comment (Amet015) [inline src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java:50]: please check identation 
    private void runCommand(String stringCommand){


    private void runCommand(String stringCommand) {

File: src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java [hunk 63]
```diff
@@ -44,61 +42,101 @@
...
         }
     }
 
     /**
-     * This method is for getting the string command.
-     *
-     * @param commandList
-     * @return command concatenated.
-     * @throws CommandValueException
+     * It is for getting the string thumbnail command.
```
Comment (Amet015) [inline src/main/java/com/jalasoft/jfc/model/video/VideoConverter.java:121]: Improve documentation, it is not necesary

File: src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoCodec.java (hunk 52 above)
Comment (Amet015) [reply src/main/java/com/jalasoft/jfc/model/command/ffmpeg/CommandVideoCodec.java:30]: please delete this line 30

**Review by Amet015 — APPROVED** [review]

**Pull request merged**